bugbug/models/rcatype.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+import numpy as np
+import xgboost
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup
+from bugbug.model import BugModel
+
+# For the moment, rca - XYZ is treated of bugtype XYZ,
+# so we don't need to store it in a dictionary.
+RCA_TYPES = [
+    "externalapifailure",
+    "logicalerror",
+    "codingerror",
+    "unhandledexceptions",
```

Comment:
We should handle subcategories more clearly to avoid mistakes, e.g. this is a sub-category of Coding Error.

I think we should have two lists, one for categories and one for subcategories, and use the first when subcategories are disabled and the second when subcategories are enabled.